src/bmi/bmi_sac.f90

```diff
@@ -1153,16 +1191,32 @@ function sac_set_int(this, name, src) result (bmi_status)
     character (len=*), intent(in) :: name
     integer, intent(in) :: src(:)
     integer :: bmi_status
+    integer(kind=int64) :: exec_status
 
     !==================== UPDATE IMPLEMENTATION IF NECESSARY FOR INTEGER VARS =================
 
     select case(name)
 !     case("model__identification_number")
 !        this%model%id = src(1)
 !        bmi_status = BMI_SUCCESS
-    case default
-       bmi_status = BMI_FAILURE
-       call write_log(" Failed to set integer value for  " // name // "", LOG_LEVEL_WARNING)
+      case("serialization_create")
+         call new_serialization_request(this%model, exec_status)
+         if (exec_status == 0) then
+            bmi_status = BMI_SUCCESS
+         else
+            bmi_status = BMI_FAILURE 
+         end if
```

Comment:
The code to call `new_serialization_request` actually needs to be in _`sac_get_int`_, with the size of the serialization pack being the output written into the provided buffer.

Reply:
Also, when that size is calculated, keep in mind that it should be `INTEGER(kind=int64)`, and the corresponding statement `serialized_size = size(model%serialization_buffer, KIND=int64)`

https://gcc.gnu.org/onlinedocs/gcc-4.4.3/gfortran/SIZE.html
https://www.intel.com/content/www/us/en/docs/fortran-compiler/developer-guide-reference/2024-2/size-function.html